src/ui/public/vislib/visualizations/point_series/area_chart.js

```diff
@@ -29,17 +29,17 @@ export default function AreaChartFactory(Private) {
    * chart
    */
   class AreaChart extends PointSeriesChart {
-    constructor(handler, chartEl, chartData, chartConfig) {
+    constructor(handler, chartEl, chartData, seriesConfigArgs) {
       super(handler, chartEl, chartData);
 
-      this.isOverlapping = (handler.visConfig.get('mode') === 'overlap');
-
+      this.seriesConfig = _.defaults(seriesConfigArgs || {}, defaults);
```

Comment:
different series dont have that much in common, so bar and line will have quite different options, thats why i am keeping this at series level .... also not that much of config is there.